### ISSUE TITLE
Failed to send alert

### ISSUE THREAD

**Issue by serversathome-personal** [ZTube/truenas-gotify-adapter#2]
Followed the exact steps laid out on TrueNAS-SCALE-24.04.2

got this error: Failed sending test alert!

container logs show this:
```

2024-08-30 15:18:27.643230-04:00======== Running on http://127.0.0.1:31662 ========
2024-08-30 15:18:27.643293-04:00(Press CTRL+C to quit)
2024-08-30 15:19:01.509381-04:00===== Alert =====
2024-08-30 15:19:01.509504-04:00TrueNAS @ TrueNAS  
2024-08-30 15:19:01.509543-04:002024-08-30T15:19:01.509543967-04:00
2024-08-30 15:19:01.509564-04:00This is a test alert
2024-08-30 15:19:01.509582-04:002024-08-30T15:19:01.509582545-04:00
2024-08-30 15:19:01.509600-04:002024-08-30T15:19:01.509600045-04:00
2024-08-30 15:19:58.457829-04:00===== Alert =====
2024-08-30 15:19:58.457885-04:00TrueNAS @ TrueNAS  
2024-08-30 15:19:58.457900-04:002024-08-30T15:19:58.457900877-04:00
2024-08-30 15:19:58.457908-04:00This is a test alert
2024-08-30 15:19:58.457914-04:002024-08-30T15:19:58.457914955-04:00
2024-08-30 15:19:58.457927-04:002024-08-30T15:19:58.457927589-04:00
2024-08-30 15:20:09.230839-04:00===== Alert =====
2024-08-30 15:20:09.230879-04:00TrueNAS @ TrueNAS  
2024-08-30 15:20:09.230885-04:002024-08-30T15:20:09.230885075-04:00
2024-08-30 15:20:09.230889-04:00This is a test alert
2024-08-30 15:20:09.230902-04:002024-08-30T15:20:09.230902197-04:00
2024-08-30 15:20:09.230908-04:002024-08-30T15:20:09.230908297-04:00
2024-08-30 15:21:27.261724-04:00===== Alert =====
2024-08-30 15:21:27.261773-04:00TrueNAS @ TrueNAS  
2024-08-30 15:21:27.261783-04:002024-08-30T15:21:27.261783507-04:00
2024-08-30 15:21:27.261789-04:00This is a test alert
2024-08-30 15:21:27.261795-04:002024-08-30T15:21:27.261795846-04:00
2024-08-30 15:21:27.261801-04:002024-08-30T15:21:27.261801388-04:00
2024-08-30 15:23:50.065857-04:00===== Alert =====
2024-08-30 15:23:50.066029-04:00TrueNAS @ TrueNAS  
2024-08-30 15:23:50.066061-04:002024-08-30T15:23:50.066061730-04:00
2024-08-30 15:23:50.066082-04:00This is a test alert
2024-08-30 15:23:50.066101-04:002024-08-30T15:23:50.066101855-04:00
2024-08-30 15:23:50.066121-04:002024-08-30T15:23:50.066121222-04:00
```

**Comment by ZTube** [comment]
It seems like the connection truenas->adapter is working but the adapter somehow cannot reach your gotify instance. I will have to implement better error handling/debug output in order to properly help you with that. Until then double check if the URL and TOKEN are correct. If you are using a domain name for your gotify instance, maybe try replacing that with the ip address of your server (In case you don't use a reverse proxy) to rule out dns issues.
I'm on TrueNAS-SCALE-24.04.2 and it works for me so it certainly is not an issue with a new truenas version.

**Comment by shuperluu** [comment]
Wild guess, but did you check "Provide access to node network namespace for the workload"?

**Comment by tomsik-radek** [comment]
> Wild guess, but did you check "Provide access to node network namespace for the workload"?

Not to hijack this issue, but I'm facing this too. I don't see any settings in the Custom App menu on Scale 24.10.0.2.
I did expose the port and I can see it running on 31662 in netstat,
The only thing in my log is 

```
2024-12-12 19:17:05.759971+00:00======== Running on http://127.0.0.1:31662 ========
2024-12-12 19:17:05.760051+00:00(Press CTRL+C to quit)
```

**Comment by shuperluu** [comment]
> > Wild guess, but did you check "Provide access to node network namespace for the workload"?
> 
> Not to hijack this issue, but I'm facing this too. I don't see any settings in the Custom App menu on Scale 24.10.0.2. I did expose the port and I can see it running on 31662 in netstat, The only thing in my log is
> 
> ```
> 2024-12-12 19:17:05.759971+00:00======== Running on http://127.0.0.1:31662 ========
> 2024-12-12 19:17:05.760051+00:00(Press CTRL+C to quit)
> ```

If you follow https://github.com/ZTube/truenas-gotify-adapter/tree/docs you don't see the checkbox?

**Comment by tomsik-radek** [comment]
> > > Wild guess, but did you check "Provide access to node network namespace for the workload"?
> > 
> > 
> > Not to hijack this issue, but I'm facing this too. I don't see any settings in the Custom App menu on Scale 24.10.0.2. I did expose the port and I can see it running on 31662 in netstat, The only thing in my log is
> > ```
> > 2024-12-12 19:17:05.759971+00:00======== Running on http://127.0.0.1:31662 ========
> > 2024-12-12 19:17:05.760051+00:00(Press CTRL+C to quit)
> > ```
> 
> If you follow https://github.com/ZTube/truenas-gotify-adapter/tree/docs you don't see the checkbox?

Yes, there is no such checkbox on ElectricEel-24.10.0.2 in IX-App. What category is it under? Env? Network?

**Comment by shuperluu** [comment]
> > > > Wild guess, but did you check "Provide access to node network namespace for the workload"?
> > > 
> > > 
> > > Not to hijack this issue, but I'm facing this too. I don't see any settings in the Custom App menu on Scale 24.10.0.2. I did expose the port and I can see it running on 31662 in netstat, The only thing in my log is
> > > ```
> > > 2024-12-12 19:17:05.759971+00:00======== Running on http://127.0.0.1:31662 ========
> > > 2024-12-12 19:17:05.760051+00:00(Press CTRL+C to quit)
> > > ```
> > 
> > 
> > If you follow https://github.com/ZTube/truenas-gotify-adapter/tree/docs you don't see the checkbox?
> 
> Yes, there is no such checkbox on ElectricEel-24.10.0.2 in IX-App. What category is it under? Env? Network?

Now I can not find that by that name. Weird.

I will check check tomorrow if "Host network" does the same. Is the only field I have at Network that I only used for this Custom App. The other charts I run doesn't use it.
![Screenshot 2024-12-16 233750](https://github.com/user-attachments/assets/8c32246e-1b35-499f-9ad0-665e91736561)

**Comment by ZTube** [comment]
I just fixed the documentation issue with "Provide access to node network namespace for the workload" being renamed to "Host Network". Concerning the original issue I added some more debug output to further diagnose the problem so please update to the latest version.

**Comment by shuperluu** [comment]
> I just fixed the documentation issue with "Provide access to node network namespace for the workload" being renamed to "Host Network". Concerning the original issue I added some more debug output to further diagnose the problem so please update to the latest version.

Should I also request to update the "docs" branch where we have the pictures?

I will update 1 one of my TrueNAS instances app today.

**Comment by ZTube** [comment]
I think I'll rather create a github wiki page for instructions. Easier to maintain and maybe a bit easier to comprehend/layout

**Comment by tomsik-radek** [comment]
I can confirm it works via a YAML deployment.
Maybe a different question, but can I call the http://localhost:31662/ endpoint with a different priority level?

**Comment by serversathome-personal** [comment]
works now. can close

**Comment by shuperluu** [comment]
> I can confirm it works via a YAML deployment. Maybe a different question, but can I call the http://localhost:31662/ endpoint with a different priority level?

You talking about this Level?
I have mine set to Info (for Gotify) and kept Warning for e-mail, but had this webhook set to Warning before and it worked.

![Image](https://github.com/user-attachments/assets/7904c7cb-c757-4575-84a7-96d80a7dee4c)

**Comment by ZTube** [comment]
> I can confirm it works via a YAML deployment.
> Maybe a different question, but can I call the http://localhost:31662/ endpoint with a different priority level?

Hi,
if you're talking about the gotify-interal priority you should be able to set a default priority when creating the application in the gotify webinterface.
Setting the priority on a per notification basis is sadly not possible at the moment

**Comment by tomsik-radek** [comment]
> > I can confirm it works via a YAML deployment.
> > Maybe a different question, but can I call the http://localhost:31662/ endpoint with a different priority level?
> 
> Hi, if you're talking about the gotify-interal priority you should be able to set a default priority when creating the application in the gotify webinterface. Setting the priority on a per notification basis is sadly not possible at the moment

Yes, I meant calling it with a different priority. Before this I used a (now dead) plugin that could accept a Slack webhook with a priority parameter. I don't work with Python but I see you already have a "priority" variable, so I guess I'm asking for this as a feature?
Either way, many thanks for this, I'm glad I don't need to rely on a native Gotify plugin!
